q2_dada2/_denoise.py

```diff
@@ -147,6 +152,60 @@ def _denoise_helper(biom_fp, track_fp, hashed_feature_ids):
              for id_ in table.ids(axis='observation')))
     return table, rep_sequences, metadata
 
+def _denoise_helper2(biom_fp, track_fp, hashed_feature_ids):
```

Comment:
This utility function is largely a duplicate of the original `_denoise_helper`, given that there is a lot of moving parts wrapped up in here, perhaps it would be safer to merge the two helpers together (at least then we won't forget to update both helpers, when making edits). It seems like most/all of the differences are related to the denoising stats code - perhaps _that_ should be factored out into its own set of utility functions?

Reply:
Yes, you are right. I have refactored ` _denoise_helper2` into the original  `_denoise_helper` function. Sorry for the mistakes.